File: Server/src/galaxy.ts
import { Config } from "./config.js";
import { Zone, type ZoneConfig } from "./zone.js";
import { ZoneType } from "./types.js";

const SOL_PROXIMA_ZONES: ZoneConfig[] = [
  {
    id: "uurf-orbit",
    name: "Uurf Orbit",
    type: ZoneType.Space,
    bounds: { xMin: -2500, xMax: 2500, yMin: -2500, yMax: 2500 },
    connections: ["uurf-hub", "uurf-surface", "asteroid-belt-alpha"],
    nodeCount: Config.nodesPerZone.default,
  },
  {
    id: "uurf-hub",
    name: "Uurf Central Hub",
    type: ZoneType.Station,
    bounds: { xMin: -1000, xMax: 1000, yMin: -1000, yMax: 1000 },
    connections: ["uurf-orbit"],
    nodeCount: 0, // stations don't have mineable resources
  },
  {
    id: "uurf-surface",
    name: "Uurf Surface - Temperate Plains",
    type: ZoneType.Planet,
    bounds: { xMin: -2000, xMax: 2000, yMin: -2000, yMax: 2000 },
    connections: ["uurf-orbit"],
    nodeCount: Config.nodesPerZone.default,
  },
  {
    id: "asteroid-belt-alpha",
    name: "Asteroid Belt Alpha",
    type: ZoneType.Space,
    bounds: { xMin: -3000, xMax: 3000, yMin: -3000, yMax: 3000 },
    connections: ["uurf-orbit"],
    nodeCount: Config.nodesPerZone.asteroidBelt,
  },
];

/** Create all zones for the Sol Proxima star system and return them as a Map keyed by zone ID. */
export function createSolProxima(): Map<string, Zone> {
  const zones = new Map<string, Zone>();
  for (const config of SOL_PROXIMA_ZONES) {
    zones.set(config.id, new Zone(config));
  }
  return zones;
}

File: ServerRS/src/game/galaxy.rs
use std::collections::HashMap;

use crate::config;

use super::zone::{Zone, ZoneBounds, ZoneConfig};

pub fn create_sol_proxima() -> HashMap<String, Zone> {
    let configs = vec![
        ZoneConfig {
            id: "uurf-orbit".to_string(),
            name: "Uurf Orbit".to_string(),
            zone_type: "space".to_string(),
            bounds: ZoneBounds {
                x_min: -2500.0,
                x_max: 2500.0,
                y_min: -2500.0,
                y_max: 2500.0,
            },
            connections: vec![
                "uurf-hub".to_string(),
                "uurf-surface".to_string(),
                "asteroid-belt-alpha".to_string(),
            ],
            node_count: config::NODES_PER_ZONE_DEFAULT,
        },
        ZoneConfig {
            id: "uurf-hub".to_string(),
            name: "Uurf Central Hub".to_string(),
            zone_type: "station".to_string(),
            bounds: ZoneBounds {
                x_min: -1000.0,
                x_max: 1000.0,
                y_min: -1000.0,
                y_max: 1000.0,
            },
            connections: vec!["uurf-orbit".to_string()],
            node_count: 0,
        },
        ZoneConfig {
            id: "uurf-surface".to_string(),
            name: "Uurf Surface - Temperate Plains".to_string(),
            zone_type: "planet".to_string(),
            bounds: ZoneBounds {
                x_min: -2000.0,
                x_max: 2000.0,
                y_min: -2000.0,
                y_max: 2000.0,
            },
            connections: vec!["uurf-orbit".to_string()],
            node_count: config::NODES_PER_ZONE_DEFAULT,
        },
        ZoneConfig {
            id: "asteroid-belt-alpha".to_string(),
            name: "Asteroid Belt Alpha".to_string(),
            zone_type: "space".to_string(),
            bounds: ZoneBounds {
                x_min: -3000.0,
                x_max: 3000.0,
                y_min: -3000.0,
                y_max: 3000.0,
            },
            connections: vec!["uurf-orbit".to_string()],
            node_count: config::NODES_PER_ZONE_ASTEROID_BELT,
        },
    ];

    let mut zones = HashMap::new();
    for config in configs {
        let id = config.id.clone();
        zones.insert(id, Zone::new(config));
    }
    zones
}
